### DISCUSSION TITLE
Export an app without the data?

### DISCUSSION THREAD

**Discussion by oswaldB** [Budibase/budibase#2869] [Help]
Hello,

Is it possible to export an app without the data?

My use case is this one. People are using the app in production.

I am developing new features for this app. 

I would like to create a copy of the app without all the data of the production one. Then to change few things on it. And at the end upload my new develop app instead of the production app

Do you think it is possible?

**Comment by mjashanks** [comment]
Hi Oswald. 

So just to check - are you using `Publish`? 

if you Publish your app, then it will be accessible on a URL like `app_<id>`.

 If don't publish your app and use click the ▶️ button to preview your app, the URL is `app_dev_<id>`.

These 2 URLs are different apps. You can make changes on the `dev` version, without affecting the published version. Once you are happy, you can "Publish" your changes. You can also "Revert" your changes to send the dev version back to the current published version.

That's how we intend you to do development.

If you know about this already - is there something about it that doesn't work for you, or makes it not useful?

**Comment by PGLange** [comment]
Hey Oswald,
you can 
- export a copy with data
- import as new app
- use database script language to clean db in docker (advanced task!!!)
- and then develop your logic with new test data .
You can kill the data within the (backuped) app by editing the downloaded *.txt file, too.

...But this only works when you are using couchDB...
For using  with sql, just copy your DB to a new DB and clean the content.

I hope this helps :)

**Reply by fuerst** [reply]
As far as I know there is no easy way to remove documents from a CouchDB database. 
I rather would do that at the exported app using an one liner like this:

```
perl -pi -e 's/{"_id":"ro_ta_.*?},//g' export.json
```

Or if you only want to purge deleted documents from the exported app:

```
perl -pi -e 's/{"_id":"ro_ta_.*?"_deleted":true.*?},//g' export.json
```

Above commands are examples only. Please make sure you test and adopt for your needs.
